File: translations/codenet/uncommented/C2Go/gpt-4o/Code/s436432602.go
package main

import (
	"fmt"
)

func run() {
	var n, k int
	fmt.Scan(&n, &k)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}
	dp := make([]int, k+1)

	for i := 1; i <= k; i++ {
		win := 0
		for j := 0; j < n && a[j] <= i; j++ {
			if dp[i-a[j]] == 0 {
				win = 1
			}
		}
		dp[i] = win
	}
	if dp[k] == 1 {
		fmt.Println("First")
	} else {
		fmt.Println("Second")
	}
}

func main() {
	run()
}

// <END-OF-CODE>

File: translations/codenet/uncommented/C2C++/gpt-4o/Code/s436432602.cpp
#include <iostream>
#include <vector>

void run() {
    int n, k;
    std::cin >> n >> k;
    std::vector<int> a(n);
    for (int i = 0; i < n; i++) std::cin >> a[i];

    std::vector<int> dp(k + 1, 0);
    for (int i = 1; i <= k; i++) {
        int win = 0;
        for (int j = 0; j < n && a[j] <= i; j++) {
            if (dp[i - a[j]] == 0) {
                win = 1;
            }
        }
        dp[i] = win;
    }
    std::cout << (dp[k] ? "First" : "Second") << std::endl;
}

int main() {
    run();
    return 0;
}

// <END-OF-CODE>
